staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -69,6 +75,17 @@ const (
 	dialTimeout = 20 * time.Second
 
 	dbMetricsMonitorJitter = 0.5
+
+	// grpcServiceConfigWithHealthcheck is used to enable etcd grpc client health check.
+	// See https://grpc.io/docs/guides/health-checking/#enabling-client-health-checking
+	//
+	// Need to specify loadBalancingPolicy because internally etcd sets `round_robin`.
+	// Using config without policy will revert to default `pick_first`
+	// https://grpc.io/docs/guides/custom-load-balancing/#implementing-your-own-policy
+	//
+	// serviceName is set to "" because etcd server uses empty string to represent the health of the whole server
+	// See https://grpc.io/docs/guides/health-checking/#the-server-side-health-service
+	grpcServiceConfigWithHealthcheck = `{"loadBalancingPolicy": "round_robin", "healthCheckConfig": {"serviceName": ""}}`
```

Comment:
> internally etcd sets `round_robin`

How do we ensure we don't drift from internally configured grpc or etcd defaults we're overriding?

> `{"loadBalancingPolicy": "round_robin", "healthCheckConfig": {"serviceName": ""}}`

How strict is the parser for this? if this config had a typo, would anything notice?

Reply:
Drift is a risk. But etcd client [resolver](https://github.com/etcd-io/etcd/blame/release-3.5/client/v3/internal/resolver/resolver.go) code hasn't changed for last 3 years.

Parser isn't strict. Will ignore typos. I've filed an [issue](https://github.com/grpc/grpc-go/issues/7818) for grpc-go library to improve it.